From b482f8705f108193b512e19cf12cdcd3b9b37478 Mon Sep 17 00:00:00 2001
From: David Bruce <davidstuartbruce@gmail.com>
Date: Wed, 9 Feb 2011 19:20:55 -0600
Subject: [PATCH 1/2] Fix of *.pc.in - only require libs actually needed

configure.ac now creates file more intelligently.  We still need to do
this for the cmake build.
---
 configure.ac     |   27 ++++++++++++++++++++++++---
 t4k_common.pc.in |   18 +++++++++++-------
 2 files changed, 35 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6ef6188..03e62b0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,6 +53,14 @@ dnl In the following checks we attempt to use pkg-config if available, falling
 dnl back to standard autoconf macro as needed:
 dnl ---------------------------------------------------------------------------
 
+dnl Create output variables to be used in t4k_common.pc.in:
+dnl NOTE - we ought to trim down the Libs.private list to secondary dependencies
+dnl that always occur even if all optional libs are deselected, and add the
+dnl others only as needed - DSB
+
+T4K_COMMON_DEP_PKGS=""
+T4K_COMMON_DEP_LIBS=""
+T4K_COMMON_DEP_LIBS_PRIVATE="-mwindows -lmikmod -lsmpeg -lstdc++ -lmingw32 -lSDLmain -liconv -luser32 -lgdi32 -lwinmm -ldxguid -lSDL -lpthread -lvorbisfile -lvorbis -lm -logg -lfreetype"
 
 
 dnl Check for SDL
@@ -65,6 +73,7 @@ PKG_CHECK_MODULES([SDL],
 
 CFLAGS="$CFLAGS $SDL_CFLAGS" 
 LIBS="$LIBS $SDL_LIBS"
+T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS sdl >= 1.2.0"
 
 
 
@@ -82,8 +91,7 @@ PKG_CHECK_MODULES([SDL_IMAGE],
 AC_DEFINE([HAVE_LIBSDL_IMAGE],[1],[Define to 1 if you have the 'SDL_image' library])
 CFLAGS="$CFLAGS $SDL_IMAGE"
 LIBS="$LIBS $SDL_IMAGE_LIBS"
-
-
+T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS SDL_image"
 
 dnl Check for SDL_mixer: --------------------------------------------------------
 
@@ -99,7 +107,7 @@ PKG_CHECK_MODULES([SDL_MIXER],
 AC_DEFINE([HAVE_LIBSDL_MIXER],[1],[Define to 1 if you have the 'SDL_mixer' library])
 CFLAGS="$CFLAGS $SDL_MIXER_CFLAGS"
 LIBS="$LIBS $SDL_MIXER_LIBS"
-
+T4K_COMMON_DEP_LIBS="$T4K_COMMON_DEP_LIBS -lSDL_mixer"
 
 
 dnl Check for SDL_Pango: --------------------------------------------------------
@@ -129,6 +137,7 @@ PKG_CHECK_MODULES([SDL_PANGO],
 AC_DEFINE([HAVE_LIBSDL_PANGO],[1],[Define to 1 if you have the `SDL_Pango` library])
 CFLAGS="$CFLAGS $SDL_PANGO_CFLAGS"
 LIBS="$LIBS $SDL_PANGO_LIBS"
+T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS SDL_Pango"
 
 fi
 
@@ -153,6 +162,7 @@ PKG_CHECK_MODULES([SDL_ttf],
 AC_DEFINE([HAVE_LIBSDL_TTF],[1],[Define to 1 if you have the `SDL_ttf` library])
 CFLAGS="$CFLAGS $SDL_TTF_CFLAGS"
 LIBS="$LIBS $SDL_TTF_LIBS"
+T4K_COMMON_DEP_LIBS="$T4K_COMMON_DEP_LIBS -lSDL_ttf"
 fi
 
 
@@ -188,6 +198,7 @@ PKG_CHECK_MODULES([SDL_NET],
 AC_DEFINE([HAVE_LIBSDL_NET],[1],[Define to 1 if you have the 'SDL_net' library])
 CFLAGS="$CFLAGS $SDL_NET_CFLAGS"
 LIBS="$LIBS $SDL_NET_LIBS"
+T4K_COMMON_DEP_LIBS="$T4K_COMMON_DEP_LIBS -lSDL_net"
 fi
 
 
@@ -207,6 +218,7 @@ PKG_CHECK_MODULES([RSVG],
 
 CFLAGS="$CFLAGS $RSVG_CFLAGS"
 LIBS="$LIBS $RSVG_LIBS"
+T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS librsvg-2.0"
 fi
 
 if test "x$with_rsvg" = xyes; then
@@ -248,6 +260,7 @@ PKG_CHECK_MODULES([XML2],
 AC_DEFINE([HAVE_LIBXML2],[1],[Define to 1 if you have the `libxml-2.0` library])
 CFLAGS="$CFLAGS $XML2_CFLAGS"
 LIBS="$LIBS $XML2_LIBS"
+T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS libxml-2.0"
 
 
 
@@ -260,6 +273,14 @@ AC_CHECK_LIB([m],
 
 
 
+dnl Output substitutions needed for t4k_common.pc.in:
+dnl (these are used by the subsequent build process of a program using t4k_common,
+dnl not by the build process of t4k_common itself).
+
+AC_SUBST(T4K_COMMON_DEP_PKGS)
+AC_SUBST(T4K_COMMON_DEP_LIBS)
+AC_SUBST(T4K_COMMON_DEP_LIBS_PRIVATE)
+
 
 # --------------------------------------------------------------------------------------
 # Checks for header files.
diff --git a/t4k_common.pc.in b/t4k_common.pc.in
index 8015913..0d6f9ea 100644
--- a/t4k_common.pc.in
+++ b/t4k_common.pc.in
@@ -1,3 +1,8 @@
+# Process with configure script to generate t4k_common.pc file for pkg-config
+# NOTE: we rely heavily on substituted output variables from Autoconf
+# or Cmake to generate a t4k_common.pc that reflects conditional build
+# with optional libraries.
+
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 libdir=@libdir@
@@ -7,13 +12,12 @@ Name: @PACKAGE_NAME@
 Description: Tux4Kids common routines
 Version: @VERSION@
 
-# AFAICT we can only list libs with .pc files in "Requires:"
-Requires: sdl >= 1.2.0 SDL_image SDL_Pango librsvg-2.0 libxml-2.0
+# List needed libs that have *.pc files in "Requires:"
+Requires: @T4K_COMMON_DEP_PKGS@ 
 # List libs "by hand" that don't have .pc files:
-Libs: -l@PACKAGE_TARNAME@ -L${libdir} -lSDL_mixer -lSDL_ttf -lSDL_net
+Libs: -l@PACKAGE_TARNAME@ -L${libdir} @T4K_COMMON_DEP_LIBS@
 Cflags: -I${includedir}
 
-# Because SDL_mixer and SDL_ttf do not have .pc files on some platforms,
-# we list their static dependencies here:
-
-Libs.private: -mwindows -L/opt/mingw-cross-env/usr/i686-pc-mingw32/lib -lmikmod -lsmpeg -lstdc++ -lmingw32 -lSDLmain -liconv -luser32 -lgdi32 -lwinmm -ldxguid -lSDL -lpthread -lvorbisfile -lvorbis -lm -logg -lfreetype
+# List secondary dependencies of libs that don't have *.pc files
+# (needed for static linking) 
+Libs.private: @T4K_COMMON_DEP_LIBS_PRIVATE@
-- 
1.7.4


From 929619c02e31d1cee0aaa6684880c476cd31764f Mon Sep 17 00:00:00 2001
From: David Bruce <davidstuartbruce@gmail.com>
Date: Thu, 10 Feb 2011 07:38:28 -0600
Subject: [PATCH 2/2] Removal of unneeded T4K_COMMON prefix from vars

Part of fix of t4k_common.pc.in
---
 configure.ac     |   28 ++++++++++++++--------------
 t4k_common.pc.in |    6 +++---
 2 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/configure.ac b/configure.ac
index 03e62b0..2d644f5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -58,9 +58,9 @@ dnl NOTE - we ought to trim down the Libs.private list to secondary dependencies
 dnl that always occur even if all optional libs are deselected, and add the
 dnl others only as needed - DSB
 
-T4K_COMMON_DEP_PKGS=""
-T4K_COMMON_DEP_LIBS=""
-T4K_COMMON_DEP_LIBS_PRIVATE="-mwindows -lmikmod -lsmpeg -lstdc++ -lmingw32 -lSDLmain -liconv -luser32 -lgdi32 -lwinmm -ldxguid -lSDL -lpthread -lvorbisfile -lvorbis -lm -logg -lfreetype"
+DEP_PKGS=""
+DEP_LIBS=""
+DEP_LIBS_PRIVATE="-mwindows -lmikmod -lsmpeg -lstdc++ -lmingw32 -lSDLmain -liconv -luser32 -lgdi32 -lwinmm -ldxguid -lSDL -lpthread -lvorbisfile -lvorbis -lm -logg -lfreetype"
 
 
 dnl Check for SDL
@@ -73,7 +73,7 @@ PKG_CHECK_MODULES([SDL],
 
 CFLAGS="$CFLAGS $SDL_CFLAGS" 
 LIBS="$LIBS $SDL_LIBS"
-T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS sdl >= 1.2.0"
+DEP_PKGS="$DEP_PKGS sdl >= 1.2.0"
 
 
 
@@ -91,7 +91,7 @@ PKG_CHECK_MODULES([SDL_IMAGE],
 AC_DEFINE([HAVE_LIBSDL_IMAGE],[1],[Define to 1 if you have the 'SDL_image' library])
 CFLAGS="$CFLAGS $SDL_IMAGE"
 LIBS="$LIBS $SDL_IMAGE_LIBS"
-T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS SDL_image"
+DEP_PKGS="$DEP_PKGS SDL_image"
 
 dnl Check for SDL_mixer: --------------------------------------------------------
 
@@ -107,7 +107,7 @@ PKG_CHECK_MODULES([SDL_MIXER],
 AC_DEFINE([HAVE_LIBSDL_MIXER],[1],[Define to 1 if you have the 'SDL_mixer' library])
 CFLAGS="$CFLAGS $SDL_MIXER_CFLAGS"
 LIBS="$LIBS $SDL_MIXER_LIBS"
-T4K_COMMON_DEP_LIBS="$T4K_COMMON_DEP_LIBS -lSDL_mixer"
+DEP_LIBS="$DEP_LIBS -lSDL_mixer"
 
 
 dnl Check for SDL_Pango: --------------------------------------------------------
@@ -137,7 +137,7 @@ PKG_CHECK_MODULES([SDL_PANGO],
 AC_DEFINE([HAVE_LIBSDL_PANGO],[1],[Define to 1 if you have the `SDL_Pango` library])
 CFLAGS="$CFLAGS $SDL_PANGO_CFLAGS"
 LIBS="$LIBS $SDL_PANGO_LIBS"
-T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS SDL_Pango"
+DEP_PKGS="$DEP_PKGS SDL_Pango"
 
 fi
 
@@ -162,7 +162,7 @@ PKG_CHECK_MODULES([SDL_ttf],
 AC_DEFINE([HAVE_LIBSDL_TTF],[1],[Define to 1 if you have the `SDL_ttf` library])
 CFLAGS="$CFLAGS $SDL_TTF_CFLAGS"
 LIBS="$LIBS $SDL_TTF_LIBS"
-T4K_COMMON_DEP_LIBS="$T4K_COMMON_DEP_LIBS -lSDL_ttf"
+DEP_LIBS="$DEP_LIBS -lSDL_ttf"
 fi
 
 
@@ -198,7 +198,7 @@ PKG_CHECK_MODULES([SDL_NET],
 AC_DEFINE([HAVE_LIBSDL_NET],[1],[Define to 1 if you have the 'SDL_net' library])
 CFLAGS="$CFLAGS $SDL_NET_CFLAGS"
 LIBS="$LIBS $SDL_NET_LIBS"
-T4K_COMMON_DEP_LIBS="$T4K_COMMON_DEP_LIBS -lSDL_net"
+DEP_LIBS="$DEP_LIBS -lSDL_net"
 fi
 
 
@@ -218,7 +218,7 @@ PKG_CHECK_MODULES([RSVG],
 
 CFLAGS="$CFLAGS $RSVG_CFLAGS"
 LIBS="$LIBS $RSVG_LIBS"
-T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS librsvg-2.0"
+DEP_PKGS="$DEP_PKGS librsvg-2.0"
 fi
 
 if test "x$with_rsvg" = xyes; then
@@ -260,7 +260,7 @@ PKG_CHECK_MODULES([XML2],
 AC_DEFINE([HAVE_LIBXML2],[1],[Define to 1 if you have the `libxml-2.0` library])
 CFLAGS="$CFLAGS $XML2_CFLAGS"
 LIBS="$LIBS $XML2_LIBS"
-T4K_COMMON_DEP_PKGS="$T4K_COMMON_DEP_PKGS libxml-2.0"
+DEP_PKGS="$DEP_PKGS libxml-2.0"
 
 
 
@@ -277,9 +277,9 @@ dnl Output substitutions needed for t4k_common.pc.in:
 dnl (these are used by the subsequent build process of a program using t4k_common,
 dnl not by the build process of t4k_common itself).
 
-AC_SUBST(T4K_COMMON_DEP_PKGS)
-AC_SUBST(T4K_COMMON_DEP_LIBS)
-AC_SUBST(T4K_COMMON_DEP_LIBS_PRIVATE)
+AC_SUBST(DEP_PKGS)
+AC_SUBST(DEP_LIBS)
+AC_SUBST(DEP_LIBS_PRIVATE)
 
 
 # --------------------------------------------------------------------------------------
diff --git a/t4k_common.pc.in b/t4k_common.pc.in
index 0d6f9ea..958fc39 100644
--- a/t4k_common.pc.in
+++ b/t4k_common.pc.in
@@ -13,11 +13,11 @@ Description: Tux4Kids common routines
 Version: @VERSION@
 
 # List needed libs that have *.pc files in "Requires:"
-Requires: @T4K_COMMON_DEP_PKGS@ 
+Requires: @DEP_PKGS@ 
 # List libs "by hand" that don't have .pc files:
-Libs: -l@PACKAGE_TARNAME@ -L${libdir} @T4K_COMMON_DEP_LIBS@
+Libs: -l@PACKAGE_TARNAME@ -L${libdir} @DEP_LIBS@
 Cflags: -I${includedir}
 
 # List secondary dependencies of libs that don't have *.pc files
 # (needed for static linking) 
-Libs.private: @T4K_COMMON_DEP_LIBS_PRIVATE@
+Libs.private: @DEP_LIBS_PRIVATE@
-- 
1.7.4

